clang/lib/CIR/CodeGen/TargetInfo.cpp

```diff
@@ -159,6 +218,66 @@ class X86_64TargetCIRGenInfo : public TargetCIRGenInfo {
 };
 } // namespace
 
+//===----------------------------------------------------------------------===//
+// Base ABI and target codegen info implementation common between SPIR and
+// SPIR-V.
+//===----------------------------------------------------------------------===//
+
+namespace {
+class CommonSPIRABIInfo : public DefaultABIInfo {
```

Comment:
@bcardosolopes @seven-mile 

Ideally, this should be moved to `clang/lib/CIR/Dialect/Transforms/TargetLowering/Targets/SPIR.cpp` unless there is some ABI-specific code generation bit that we want to have in CIR for analysis.  This would centralize target-lowering logic speeding up the `TargetLowering` library progress and defer any convoluted calling convention scenarios in SPIR to the `CallConvLoweringPass`.

It might not be a simple change, though, as it would required:

 - Priming the `TargetLowering` library for the SPIR target.
 - Adding a `CallConvAttr` attribute in `FuncOp`/`CallOp`s to carry the CC information (currently we assume everything to be the default CC).
 - Add code path for lowering SPIR calling convention for both `FuncOp`/`CallOp`s in `TargetLowering` lib.

Reply:
Thanks for suggestions! I think it matches my anticipation overall.

> Direct is fine, but extend might be doing more that it should for now, let's wait on @sitio-couto to coordinate how to go about this so you both are in sync.

Could you please further provide what should I do for Extend ArgInfo? Let me not make changes beyond your current plan.

Reply:
If your tests require `Extend`, just implement the bare minimum required for them to work, while ensuring any other code path raises an exception. If you're not currently using it in your tests, you can just add a `MissingFeatures` marker wherever they are used.

Overall, I think this is fine. It shouldn't be too difficult merging it with the `TargetLowering` library/